File: app/assets/javascripts/models/station.ts
export interface IStation {
    id: string;
    time: number;
    name: string;
    consoleOptions: string[];
    console: string;
    game: string;
}

export class Station implements IStation {

    public id: string;

    public time: number;

    public name: string;

    public consoleOptions: string[];

    public console: string;

    public game: string;

    constructor(id: string, time: number, name: string, consoleOptions: string[], console: string, game: string) {
        this.id = id;
        this.time = time;
        this.name = name;
        this.consoleOptions = consoleOptions;
        this.console = console;
        this.game = game;
    }

    static fromJSON(obj: IStation): Station {
        return $.extend({}, obj)
    }
}

File: app/models/Station.scala
package models

import store.StationStore

import scala.concurrent.duration.Duration

case class Station(var id: String, var time: Float = 0f, var name: String = "", var consoleOptions: Array[String], var console: String = "", var game: String = "") { //TODO add state (running, paused, in tournament, out of time, etc.)

}

object Station {

    val MAX_STATION_TIME: Float = 3600000f

    def tick(time: Duration, id: String)(implicit stationStore: StationStore): Unit = {
        val stationOpt: Option[Station] = stationStore.get(id)
        if (stationOpt.isDefined) {
            val station: Station = stationOpt.get
            station.time = Math.max(station.time - time.toMillis, 0)
        }
    }
}
